Feat/#327/ml page

AlvinCao88 opened the pull request:
# Description

<!-- Please include a summary of the change -->
New Page  for ML program bootcamp:
- Link to page from the NAvbar main page
- Different sections: about the event
- timeline or outline of the event
- content to be displayed and shared

Relevant GitHub Issues that will be fixed are New ML Program Page #327

## Type of change

<!-- Select all options that are relevant. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update
- [ ] Chore (anything else)

## How Has This Been Tested?

<!-- Summary of tests conducted -->

- [ ] Responsiveness on various devices
- Desktop Browser
  - [ ] Apple Safari
  - [x] Google Chrome
  - [x] Microsoft Edge
  - [x] Mozilla FireFox
- Mobile Browser
  - [ ] Apple Safari
  - [x] Google Chrome
  - [x] Microsoft Edge
  - [x] Mozilla FireFox
- Other Tests:

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## PR Closure

- Use **Squash And Merge** to Close Pull Request
- Delete any branch after merging

AlvinCao88 committed:
feat: ✨ Implemented new ML Page

Linked from the home page on the NavBar, the Page include es the description, Outline and Content, and a link back to the home page.

#327

AlvinCao88 committed:
style: 🎨 Adjusted styles and Better names for sections

Included the ML image, improved the navbar routing, changed name for the ml tab and Adjusted styles for the page.

AlvinCao88 committed:
style: 🎨 Changed the name for about

changed the name for about section

AlvinCao88 committed:
fix: 🐛 Lint fix

Changed quote and double quotes for lint

AlvinCao88 committed:
style: 🎨 Formatted Code

ThunderRoar commented on src/components/ML/About/index.tsx:
```diff
@@ -0,0 +1,28 @@
+  return (
+    <section className="flex flex-col items-center justify-center space-y-10 text-white">
+      <div className="flex flex-col gap-6 xl:gap-12">
+        <h1 className="text-center text-white text-4xl font-medium tracking-widest md:text-5xl lg:text-7xl">
```
Small change for consistency from the home page title colouring. Use instead text-[#d9d9d9].
Similarly to the other sections of this page.

ThunderRoar reviewed (changes requested):
For the Content section, it would look nice to have future lessons greyed out or have a  comment "Coming soon" instead of "no recordings/slides available"

meliadamian17 commented on src/app/ml/page.tsx:
```diff
@@ -0,0 +1,23 @@
+const links: Link[] = [
+  { title: 'Outline', href: '/ml#outline' },
+  { title: 'Content', href: '/ml#content' },
+  { title: 'Home', href: '/#intro' },
```
Could We make the `Home` be first in the list? I feel like that's most common UI practices

meliadamian17 commented on src/components/ML/Content/index.tsx:
```diff
@@ -0,0 +1,90 @@
+            <h3 className="text-2xl font-bold mb-4">{`Lesson ${index + 1} on ${lesson.date}`}</h3>
+            <p className="mb-2">
+              {lesson.recording ? (
+                <a
```
Could we change these using NextJS built-in `<Link>`?

ThunderRoar commented on src/app/page.tsx:
```diff
@@ -12,6 +12,7 @@ const links: Link[] = [
   { title: 'Sponsors', href: '/#sponsors' },
   { title: 'Team', href: '/#team' },
   { title: 'Datathon', href: '/datathon' },
+  { title: 'ML Program', href: '/ml' },
```
Change '/ml' to '/ml-program' for a descriptive url path.

AlvinCao88 committed:
style: 🎨 changed code based on PR Comments

Made some changes from PR comments, the style of content, ml-page url,navbar links and links to recordings and slides

AlvinCao88 committed:
Merge remote-tracking branch 'refs/remotes/origin/feat/#327/ml-page' into feat/#327/ml-page

ThunderRoar committed:
fix: 🚑️ home page contact email

ThunderRoar reviewed (approved):
Looks good
